common/src/region.rs

```diff
@@ -282,20 +252,35 @@ impl RegionDefinition {
      */
     pub fn validate_io(
         &self,
-        offset: Block,
+        offset: BlockIndex,
         length: usize,
+        ddef: &RegionDefinition,
     ) -> Result<(), CrucibleError> {
-        if offset.shift != self.extent_size.shift {
-            return Err(CrucibleError::BlockSizeMismatch);
-        }
-
-        let final_offset = offset.byte_value() + length as u64;
+        let final_offset = offset.0 * ddef.block_size + length as u64;
```

Comment:
Why not use `self.block_size` here? I don't grok why there's a `RegionDefinition` argument now.

Reply:
lol no idea what I was doing, fixed